### ISSUE TITLE
Mapping lists resulting in empty list

### ISSUE THREAD

**Issue by rsoffiatto** [dobrynya/nomin#21]
Hi,
I'm trying to map two classes that have Lists attributes. If one of this Lists are empty I have to get an empty List on the output object. The problem is, the output object always have a null instead an empty list when the input object has an empty list on the mapped attribute.

Simplified example:

```Java
public class A {
    private List<C> cList;

    ...
}

public class B {
    private List<D> dList;

    ...
}
```

```Groovy
mappingFor a:A, b:B

b.cList = a.dList
```

```Java
B b = new B();
b.dList = new ArrayList<>();
A a = nomin.map(b, A.class);

assert a.cList != null //false
```

I tried to map this in different ways. Tried defining two different one-way rules with something like:

```Groovy
a.cList = { b.dList.size() == 0 ? [] : mapper.map(b.dList, C.class) }
```
but the result is always the same.

It seems the change is made in CollectionRuleElem::set, line 24. If the source.size() == 0 then the property is set to null. 
Is there a way to create a mapping so empty lists are mapped to empty lists? I was not able to find one.

**Comment by dobrynya** [comment]
Hi rsoffiatto

It was a bit of an optimization :) More over there is people considering this is right behaviour. I was considering so some time ago. Now I'm not.

Well. I can advise to use **after** hook and set an empty list to a property in case of null.

**Comment by ollio** [comment]
Hi dobrynya,

I have also a problem with the solution to set a empty list to null. In one case a setter of the target object throws a NPE and breaks the mapping. 
I have written a patch which is solving the problem. Can you review it and attach it to the master?

The better solution for the future is to make the behavior of the Collection-mapper configurable.

**Comment by dobrynya** [comment]
The required will be available in release 1.1.4. I've released it today. Please let me know in case of any problem
